lib/faraday/error.rb

```diff
@@ -74,35 +98,35 @@ class ServerError < Error
   end
 
   # A unified client error for timeouts.
-  class TimeoutError < ServerError
+  class TimeoutError < ClientError
     def initialize(exc = 'timeout', response = nil)
       super(exc, response)
     end
   end
 
   # Raised by Faraday::Response::RaiseError in case of a nil status in response.
   class NilStatusError < ServerError
-    def initialize(_exc, response: nil)
-      message = 'http status could not be derived from the server response'
-      super(message, response)
+    def initialize(exc, response = nil)
+      exc_msg_and_response!(exc, response)
+      super('http status could not be derived from the server response')
     end
   end
 
   # A unified error for failed connections.
-  class ConnectionFailed < Error
+  class ConnectionFailed < ClientError
```

Comment:
I previously changed the superclass of these classes because it made more sense.
These changes have been reflected on the `UPGRADING.md` so people should be aware.
Error classes are a bit messy in Faraday, 1.0 is our only chance to change that while making people aware.
If we want to leave them as they are then this will need to wait 2.0.
Do you think it will be too much disruptive?

Reply:
Doh, I'll revert those changes. It was a misunderstanding of mine.